crates/astria-sequencer-relayer/src/config.rs

```diff
@@ -38,18 +54,105 @@ pub struct Config {
 }
 
+fn create_filter(input: &str) -> eyre::Result<HashSet<RollupId>> {
+    input
+        .split(',')
+        .filter(|base64_encoded_id| !base64_encoded_id.is_empty())
+        .map(|base64_encoded_id| {
+            BASE64_STANDARD
+                .decode(base64_encoded_id.trim())
+                .wrap_err_with(|| {
+                    format!(
+                        "failed to base64-decode rollup id '{base64_encoded_id}' in configured \
+                         rollup_id_filter"
+                    )
+                })
+                .and_then(|raw_id| {
+                    RollupId::try_from_slice(&raw_id).wrap_err_with(|| {
+                        format!(
+                            "failed to parse '{base64_encoded_id}' as a rollup id in configured \
+                             rollup_id_filter"
+                        )
+                    })
+                })
+        })
+        .collect()
+}
+
 #[cfg(test)]
 mod tests {
-    use super::Config;
+    use astria_core::primitive::v1::RollupId;
+    use itertools::Itertools;
+
+    use super::*;
 
     const EXAMPLE_ENV: &str = include_str!("../local.env.example");
 
     #[test]
     fn example_env_config_is_up_to_date() {
         config::tests::example_env_config_is_up_to_date::<Config>(EXAMPLE_ENV);
     }
+
+    #[test]
+    fn should_create_filter() {
+        let rollup_ids: HashSet<_> = (0..10).map(|i| RollupId::new([i; 32])).collect();
+
+        // Normal form: "aaa,bbb,ccc".
+        let input = rollup_ids.iter().join(",").to_string();
+        assert_eq!(create_filter(&input).unwrap(), rollup_ids);
+
+        // With trailing comma: "aaa,bbb,ccc,".
+        let input = format!("{},", rollup_ids.iter().join(","));
+        assert_eq!(create_filter(&input).unwrap(), rollup_ids);
+
+        // With extra commas: "aaa,,bbb,,ccc,,".
+        let input = format!("{},,", rollup_ids.iter().join(",,"));
+        assert_eq!(create_filter(&input).unwrap(), rollup_ids);
+
+        // With spaces after commas: "aaa, bbb, ccc".
+        let input = rollup_ids.iter().join(", ").to_string();
+        assert_eq!(create_filter(&input).unwrap(), rollup_ids);
+
+        // With spaces before and after commas: "aaa , bbb , ccc".
+        let input = rollup_ids.iter().join(" , ").to_string();
+        assert_eq!(create_filter(&input).unwrap(), rollup_ids);
+
+        // Single entry: "aaa".
+        let single_id = RollupId::new([100; 32]);
+        let input = single_id.to_string();
+        assert_eq!(
+            create_filter(&input).unwrap(),
+            std::iter::once(single_id).collect(),
+            "{input}"
+        );
+
+        // No entries: "".
+        assert!(create_filter("").unwrap().is_empty());
+    }
+
+    #[test]
+    fn should_fail_to_create_filter_from_bad_input() {
+        // Invalid base64 encoding.
+        let input = "CAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAg!";
+        assert!(create_filter(input).is_err());
```

Comment:
My preference for the sort of tests is to `unwrap_err` instead of asserting on a boolean. If this fails then the person for whom this failed (i.e. `create_filter` returned with success) will have to start rewriting the test to figure out why this is `false`

Reply:
Done in [aa8957e](https://github.com/astriaorg/astria/pull/1001/commits/aa8957e8c1a7a2d49742e755f3c4b0617fbda1e9).